ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -118,6 +119,144 @@ private class JdbcLedgerDao(
+          val participantId = LedgerString
+            .fromString(participantIdRaw)
+            .fold(
+              err => sys.error(s"Failed to decode participant id in configuration entry: $err"),
+              identity)
+
+          offset ->
+            (typ match {
+              case "accept" =>
+                ConfigurationEntry.Accepted(
+                  submissionId = submissionId,
+                  participantId = participantId,
+                  configuration = config
+                )
+              case "reject" =>
+                ConfigurationEntry.Rejected(
+                  submissionId = submissionId,
+                  participantId = participantId,
+                  rejectionReason = rejectionReason.getOrElse("<missing reason>"),
+                  proposedConfiguration = config
+                )
+
+              case _ =>
+                sys.error(s"getConfigurationEntries: Unknown configuration entry type: $typ")
+            })
+      }
+
+  override def getConfigurationEntries(
+      startInclusive: Long,
+      endExclusive: Long): Source[(Long, ConfigurationEntry), NotUsed] =
+    paginatingStream(
+      startInclusive,
+      endExclusive,
+      PageSize,
+      (startI, endE) => {
+        Source
+          .fromFuture(dbDispatcher.executeSql(s"load ledger entries [$startI, $endE[") {
+            implicit conn =>
+              SQL_GET_CONFIGURATION_ENTRIES
+                .on("startInclusive" -> startI, "endExclusive" -> endE)
+                .as(configurationEntryParser.*)
+          })
+          .flatMapConcat(Source(_))
+      }
+    )
+
+  private val SQL_INSERT_CONFIGURATION_ENTRY =
+    SQL(
+      """insert into configuration_entries(ledger_offset, recorded_at, submission_id, participant_id, typ, rejection_reason, configuration)
+        |values({ledger_offset}, {recorded_at}, {submission_id}, {participant_id}, {typ}, {rejection_reason}, {configuration})
+        |""".stripMargin)
+
+  override def storeConfigurationEntry(
+      offset: LedgerOffset,
+      newLedgerEnd: LedgerOffset,
+      externalOffset: Option[ExternalOffset],
+      recordedAt: Instant,
+      submissionId: String,
+      participantId: ParticipantId,
+      configuration: Configuration,
+      rejectionReason: Option[String]
+  ): Future[PersistenceResponse] = {
+    dbDispatcher.executeSql(s"store configuration entry submissionId=$submissionId") {
+      implicit conn =>
+        // If the entry is stored by indexer, we never expect the generation or duplicate submission
+        // and the indexer should correctly crash on duplicate. If we're storing for sandbox, then we can
+        // expect this to fail with "Duplicate".
+        val currentConfig = selectLedgerConfiguration
+        if (rejectionReason.isEmpty && (currentConfig exists (_.generation + 1 != configuration.generation))) {
+          PersistenceResponse.Duplicate
```

Comment:
We still want to update the ledger end in this case.

Reply:
:+1: